### DISCUSSION TITLE
Helix fails to redraw on Linux console

### DISCUSSION THREAD

**Discussion by atagen** [helix-editor/helix#10408] [Troubleshooting]
### Summary

When using helix from Linux console in certain environments (not terminal emulators, actual kernel console) it doesn't appear to screen clear properly on open. Every non-occupied character space is still filled with whatever junk was at that position already. This leaves hx almost unusable unless you spam the screen to forcibly clear all those positions.

First encountered on NixOS's install ISO. Annoyingly, this doesn't appear to reproduce on an installed machine.
While a redraw command (as discussed in issue #4350) would solve the issue, this inability to reproduce in an installed environment leads me to believe this might be worth its own bug report.

Other editors tested such as Neovim, Vim, Nano, etc. appear to work fine in this case.

I understand this is a bit of an esoteric issue, please let me know if there's something I can do to better demonstrate or inform on this.

### Reproduction Steps

I tried this:

1. `hx /etc/nixos/configuration.nix`

I expected this to happen:
An editor opens containing only the contents of the file

Instead, this happened:
A big mess of scrollback and code

### Platform

Linux

### Terminal Emulator

None

### Helix Version

22.12

**Comment by CptPotato** [comment]
I ran into this issue a while ago, but assumed I just didn't configure something correctly. In the end I ignored it since I don't use helix outside of some desktop environment or window manager anymore.

I think I can still reliably reproduce this, though.

**Comment by icecreammatt** [comment]
I'm running into this too. I bound a hotkey to open lazygit when I do <Leader>g which works but then when quitting the screen is partially in a bad state. I'd like a screen redraw to occur after the run-shell-command process completes.

This is happening on Mac in WezTerm

**Reply by BastienVigneron** [reply]
Idem for me

**Comment by DyBev** [comment]
Can confirm still occurs, not immediately understood that it's an issue with the console vs helix, might be worth noting it somewhere until there is a fix / workaround for it.
Fedora is the OS .

**Comment by BrayanLope5** [comment]
Having the same problem. I am using Arch with hyprland, kde, gnome, and all of them have the same problem. If there is some other tui or program that outputs text while helix is running, I get some of that output on the screen but in a huge mess. I also get this problem on both, wezterm and alacritty. I've used bash, zsh, and fish and same problem. I have to run `:redraw` every time.

**Comment by silverhairs** [comment]
Does moving this from an issue to a discussion means that it got deprioritized?

**Reply by atagen** [reply]
it's probably because :redraw got implemented, iirc it wasn't back then
